### PR TITLE
Add production notebook product

### PR DESCRIPTION
**Pull request by tthyer**
Adds product and template for a Linux notebook product.
Currently the only choice for "Notebook Type" is Rstudio. Eventually we'll add Jupyter as an option too.

Note, this is the production version. There is a separate PR for the development version: https://github.com/Sage-Bionetworks/scipoolprod-sc-lib-infra/pull/90

### CODE REVIEW

File: ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml [hunk 1]
```diff
@@ -0,0 +1,500 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: 'Service Catalog: EC2 Reference Architecture.'
```
Comment (zaro0508) [inline ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml]: shouldn't this mention something about notebook?

File: ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+  NotebookType:
+    Type: String
+    Description: Type of notebook software to install
+    Default: Rstudio
+    AllowedValues:
+      - Rstudio
+  VolumeSize:
+    Description: The EC2 volume size (in GB)
+    Type: Number
+    Default: 8
```
Comment (zaro0508) [inline ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml]: is this a good default and min for a notebook?  default should probably be larger for rstudio, no?

File: ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+      NotebookType:
+        default: Notebook Type
+      VolumeSize:
+        default: Disk Size
+      PrivateNetwork:
+        default: Use private network?
+Mappings:
+  NotebookTypes:
+    Rstudio:
+      AMIID: ami-036688a77595e31cf
```
Comment (zaro0508) [inline ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml]: maybe add a comment and refer to the build of this AMI?  https://travis-ci.org/Sage-Bionetworks/packer-rstudio/builds/658411244

File: ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml [hunk 4]
```diff
@@ -0,0 +1,500 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: 'Service Catalog: EC2 Reference Architecture.'
+Metadata:
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+      - Label:
+          default: Linux Instance Configuration
+        Parameters:
+          - EC2InstanceType
+          - LinuxDistribution
```
Comment (zaro0508) [inline ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml]: i don't see this in the Parameters section of template?

File: ec2/sc-portfolio-ec2.yaml [hunk 5]
```diff
@@ -127,7 +127,7 @@ Resources:
       PrincipalARN: !Sub 'arn:aws:iam::${AWS::AccountId}:group/${PrincipalGroupName2}'
       PortfolioId: !Ref 'SCEC2portfolio'
       PrincipalType: IAM
-  ec2linuxjumpcloudproduct:
+  Ec2LinuxJumpcloudProduct:
```
[the remark this answers is not in the document]
Comment (tthyer) [reply ec2/sc-portfolio-ec2.yaml]: They are already the same

File: ec2/sc-portfolio-ec2.yaml [hunk 6]
```diff
@@ -141,7 +141,21 @@ Resources:
...
+        LaunchConstraintARN: !If
+          - CreateLaunchConstraint
+          - !GetAtt 'LaunchConstraintRole.Outputs.LaunchRoleArn'
+          - !Sub 'arn:aws:iam::${AWS::AccountId}:role/${LaunchRoleName}'
+        PortfolioId: !Ref 'SCEC2portfolio'
+        RepoRootURL: !Ref 'RepoRootURL'
+        StackDatetime: !Ref StackDatetime
+      TemplateURL: !Sub '${RepoRootURL}ec2/sc-product-ec2-linux-jumpcloud-notebook.yaml'
+      TimeoutInMinutes: 5
+  Ec2WindowsJumpcloudProduct:
```
[the remark this answers is not in the document]
Comment (tthyer) [reply ec2/sc-portfolio-ec2.yaml]: Again, that change was already made, see PR #90

File: ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml (hunk 1 above)
Comment (tthyer) [reply ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml]: None of the existing EC2 templates do this, but sure, I can make that change to all the templates in this PR

File: ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml (hunk 4 above)
Comment (tthyer) [reply ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml]: Should have been NotebookType. Fixed.

File: ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml (hunk 3 above)
Comment (tthyer) [reply ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml]: done

File: ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml (hunk 2 above)
Comment (tthyer) [reply ec2/sc-ec2-linux-jumpcloud-notebook-v1.0.0.yaml]: Raised to 16 based on VolumeSize in packer-rstudio template.json

**Pull request merged**